### ISSUE TITLE
Support authentication when Toolpad app is imported as an iframe

### ISSUE THREAD

**Issue by prakhargupta1** [mui/mui-toolpad#3223]
### Search keywords

iframe, authentication

### Latest version

- [X] I have tested the latest version

### Summary

As reported here: https://github.com/mui/mui-private/pull/376#issuecomment-1942905491 (point-3), it is to be explored if Toolpad app when used as iframe can take the authentication info from a 3rd party app like Zendesk and run seamlessly?

**Comment by apedroferreira** [comment]
Oh, I can also probably make the login screen show in iframes if that's good enough, and people show be able to log in inside the iframe?

**Comment by prakhargupta1** [comment]
> Did the app work in iframes with basic authentication before or was it unprotected?
Yes, a page from tools-private works in Zendesk with basic authentication. For Google authentication, asking users to log in could be tricky, as discussed [here](https://community.retool.com/t/trouble-embedding-my-retool-app-into-an-iframe/2818/5).

**Comment by Janpot** [comment]
I'd prefer this to be on hold until the security implications are well understood
